### PR TITLE
Material synth whitelist

### LINKED ISSUE #16901: Engineering cyborgs can scan and produce leather
#### Steps to reproduce 
Obtain leather
Scan leather
Print leather

The latest changes to the mat synth probably caused this.

#### Issue bingo

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

### PR DESCRIPTION
**Pull request by Probe1**
Material synthesizers can now only scan the intended items.  Engineering borgs are blocked from scanning bananium and phazon in a new way.  

Whitelist:
/obj/item/stack/sheet/**metal**
/obj/item/stack/sheet/**glass**/,
/obj/item/stack/sheet/**wood**,
/obj/item/stack/sheet/**plasteel**,
/obj/item/stack/sheet/**mineral**
Mineral includes gold, plasma, plastic, phazon, et cetera.  If there's anything on this list you felt should have been included feel free to shout it out and call me an evil bastard destroying vgstation.

Resolves https://github.com/vgstation-coders/vgstation13/issues/16901

🆑 

* bugfix: Material synthesizers can no longer scan stuff like leather or cardboard.

### CODE REVIEW

**Comment by Kurfursten** [comment]
In what sort of situation would you need to scan metal, glass, or plasteel?

**Comment by Probe1** [comment]
In cases where someone does something stupid in the future.  Say JojoFan97 makes a matsynth/matsynth_empty then someone makes an issue bitching that they can't scan metal.

Glass is needed to scan rglass, plasmaglass, et cetera.

File: code/game/objects/items/devices/mat_synth.dm [hunk 1]
```diff
@@ -36,7 +42,7 @@
 
 /obj/item/device/material_synth/robot/engiborg/New() //We have to do this during New() because BYOND can't pull a typesof() during compile time.
 	. = ..()
-	can_scan = existing_typesof(/obj/item/stack/sheet) - list(/obj/item/stack/sheet/mineral/clown, /obj/item/stack/sheet/mineral/phazon)
+	var/list/cant_scan = list(/obj/item/stack/sheet/mineral/clown, /obj/item/stack/sheet/mineral/phazon)
```
Comment (DamianX) [inline code/game/objects/items/devices/mat_synth.dm]: this is a local var, remove the `var/list/` part to make it what you intended

**Review by DamianX — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/mat_synth.dm (hunk 1 above)
Comment (Probe1) [reply code/game/objects/items/devices/mat_synth.dm]: I hate you dreammaker.

**Comment by DamianX** [comment]
Why are we not just removing entries from `can_scan` anyway?

**Comment by ghost** [comment]
@DamianX can_scan is compared with is_type_in_list (which includes children). If you want to include all types of /obj/item/stack/sheet/mineral EXCEPT bananium and phazon this is a much better way of doing it

**Review by DamianX — APPROVED** [review]

**Pull request merged**